awx/main/management/commands/run_dispatcher.py

```diff
@@ -116,9 +52,8 @@ def handle(self, *arg, **options):
         # for the DB and memcached connections (that way lies race conditions)
         django_connection.close()
         django_cache.close()
-        beat = Process(target=self.beat)
-        beat.daemon = True
-        beat.start()
+
+        scheduler.run_continuously()
```

Comment:
So this method itself doesn't run forever. Execution continues to down below where `consumer.run()` _does_ run continuously (please correct me if I'm wrong).

As to how this runs the scheduler but still gives execution control back...

https://schedule.readthedocs.io/en/stable/faq.html#how-to-continuously-run-the-scheduler-without-blocking-the-main-thread

> Run the scheduler in a **separate thread**. Mrwhick wrote up a nice solution in to this problem here (look for run_continuously())

(emphasis mine)

https://github.com/mrhwick/schedule/pull/1

> The scheduler should then call run_pending on its own, you do not need to call it explicitly in a loop.
> To stop the scheduler, you can do:
> cease.set()

Do we need to do anything like that in the same general area that we do `consumer.stop()`? If we do need some cleanup like this, we have a clear place it would go, but there's much more I don't know about joining threads in python than what I do know.

Reply:
okay, the gap between `run_pending` and `run_continuously` isn't a whole lot

https://github.com/mrhwick/schedule/blob/8e1d5f806d34d9ecde3c068490c8d1513ed774c3/schedule/__init__.py#L76-L87

```python
        cease_continuous_run = threading.Event()

        class ScheduleThread(threading.Thread):
            @classmethod
            def run(cls):
                while not cease_continuous_run.is_set():
                    self.run_pending()
                    time.sleep(interval)

        continuous_thread = ScheduleThread()
        continuous_thread.start()
        return cease_continuous_run
```

And the interval is just 1. You _could_ do a smarter interval, using the greatest common divisor of the schedule intervals....

Reply:
I'm missing something, why would they use an interval of 1 second, when they have methods specifically for computing the next run time like `_schedule_next_run`?? Even our old implementation did this IIRC

Reply:
🤷‍♂ https://github.com/mrhwick/schedule/compare/master...AlanCoding:sleep_interval?expand=1

Reply:
> Do we need to do anything like that in the same general area that we do consumer.stop()

Nah, it's just a thread.  It'll keep running forever until the main process exits.

> And the interval is just 1. You could do a smarter interval, using the greatest common divisor of the schedule intervals...

Yep, that's fair.  PR updated.